sql/sql_acl.cc

```diff
@@ -13183,14 +13183,19 @@ struct MPVIO_EXT :public MYSQL_PLUGIN_VIO
 /**
   a helper function to report an access denied error in most proper places
 */
-static void login_failed_error(THD *thd)
+static void login_failed_error(THD *thd, uint err_override = ER_ACCESS_DENIED_ERROR)
```

Comment:
I tend to dislike default values on arguments as its easy to get wrong. Can you change the other callers of `login_failed_error` to pass `ER_ACCESS_DENIED` rather than the default being here.